modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase

```diff
@@ -435,6 +597,89 @@ definition {
 		}
 	}
 
+	@description = "COMMERCE-11611. As a supplier account user, I cannot view the products in a catalog the account is not linked to."
+	@priority = 4
+	test CannotViewProductsInACatalogNotLinkedToSupplierAccount {
+		property portal.acceptance = "true";
+
+		task ("Given a Supplier account with a supplier user") {
+			JSONAccountEntry.addAccountEntry(
+				accountEntryName = "Commerce Supplier",
+				accountEntryType = "Supplier");
+
+			JSONAccountEntryUser.addAccountEntryUser(
+				accountEntryName = "Commerce Supplier",
+				userEmailAddress = "supplier@liferay.com",
+				userFirstName = "Supplier",
+				userLastName = "User",
+				userScreenName = "supplieruser");
+
+			JSONUser.setFirstPassword(
+				agreeToTermsAndAnswerReminderQuery = "true",
+				requireReset = "false",
+				userEmailAddress = "supplier@liferay.com");
+
+			JSONAccountRole.associateUser(
+				accountEntryName = "Commerce Supplier",
+				accountRole = "Account Supplier",
+				userEmailAddress = "supplier@liferay.com");
+		}
+
+		task ("And Supplier account is linked to Minium catalog") {
+			var accountId = CommerceJSONAccountsAPI.getAccountFieldByAccountName(
+				accountName = "Commerce Supplier",
+				field = "id");
+
+			CommerceJSONCatalogsAPI._patchCommerceCatalog(
+				accountId = ${accountId},
+				catalogName = "Minium");
+		}
+
+		task ("And a new Catalog with a product is added") {
+			CommerceJSONCatalogsAPI._addCatalog(catalogName = "Test Catalog");
+
+			CommerceJSONProductsAPI._addCommerceProduct(
+				catalogName = "Test Catalog",
+				price = 5,
+				productName = "Simple Product",
+				productType = "simple",
+				sku = "TESTSKU");
+		}
+
+		task ("And a product is added to the Master catalog") {
+			CommerceJSONProductsAPI._addCommerceProduct(
+				catalogName = "Master",
+				price = 25,
+				productName = "New Product",
+				productType = "simple",
+				sku = "NEWSKU");
+		}
+
+		task ("When the supplier user logs in") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "supplier@liferay.com",
+				userLoginFullName = "Supplier User");
+		}
+
+		task ("Then cannot view Simple Product from Test Catalog") {
+			CommerceProducts.openProductsAdmin();
+
+			CommerceNavigator.searchEntry(entryName = "Simple Product");
+
+			AssertElementNotPresent(
+				key_entryName = "Simple Product",
+				locator1 = "CommerceNavigation#ENTRY_NAME");
+		}
+
+		task ("And also New Product from Master Catalog") {
+			CommerceNavigator.searchEntry(entryName = "New Product");
+
+			AssertElementNotPresent(
+				key_entryName = "New Product",
+				locator1 = "CommerceNavigation#ENTRY_NAME");
+		}
+	}
```

Comment:
Can you associate the supplier with the catalog you created and change your assertions accordingly? Thanks!